cloudify_agent/operations.py

```diff
@@ -206,54 +183,157 @@ def create_new_agent_dict(old_agent):
     return new_agent
 
 
-def create_agent_from_old_agent(install_agent_timeout=300):
-    if 'cloudify_agent' not in ctx.instance.runtime_properties:
-        raise NonRecoverableError(
-            'cloudify_agent key not available in runtime_properties')
-    old_agent = ctx.instance.runtime_properties['cloudify_agent']
-    new_agent = create_new_agent_dict(old_agent)
+@contextmanager
+def _celery_client(ctx, agent):
     # We retrieve broker url from old agent in order to support
     # cases when old agent is not connected to current rabbit server.
-    broker_url = _get_broker_url(ctx, old_agent)
-    env_broker_url = os.environ.get('CELERY_BROKER_URL')
-    os.environ['CELERY_BROKER_URL'] = broker_url
+    if 'broker_config' in agent:
+        broker_config = agent['broker_config']
+    else:
+        broker_config = ctx.bootstrap_context.broker_config()
+    broker_url = utils.internal.get_broker_url(broker_config)
+    ctx.logger.info('Connecting to {0}'.format(broker_url))
+    celery_client = celery.Celery()
+    # We can't pass broker_url to Celery constructor because it would
+    # be overriden by the value from broker_config.py.
+    config = {
+        'BROKER_URL': broker_url,
+        'CELERY_RESULT_BACKEND': broker_url
+    }
+    if agent['version'] != '3.2':
+        config['CELERY_TASK_RESULT_EXPIRES'] = \
+            defaults.CELERY_TASK_RESULT_EXPIRES
+    fd, cert_path = tempfile.mkstemp()
+    os.close(fd)
     try:
-        celery_client = celery.Celery(broker=broker_url, backend=broker_url)
-        if new_agent['old_agent_version'] != '3.2':
-            celery_client.conf.update(
-                CELERY_TASK_RESULT_EXPIRES=defaults.CELERY_TASK_RESULT_EXPIRES
-            )
+        if broker_config.get('broker_ssl_enabled'):
+            with open(cert_path, 'w') as cert_file:
+                cert_file.write(broker_config.get('broker_ssl_cert', ''))
+            broker_ssl = {
+                'ca_certs': cert_path,
+                'cert_reqs': ssl.CERT_REQUIRED
+            }
+        else:
+            broker_ssl = False
+        config['BROKER_USE_SSL'] = broker_ssl
+        celery_client.conf.update(**config)
+        yield celery_client
+    finally:
+        os.remove(cert_path)
+
+
+def _assert_agent_alive(name, celery_client):
+    tasks = utils.get_agent_registered(name, celery_client)
+    if not tasks:
+        raise NonRecoverableError(
+            'Could not access tasks list for agent {0}'.format(name))
+    if 'script_runner.tasks.run' not in tasks:
+        raise NonRecoverableError(
+            ('Task script_runner.tasks.run is not available in agent '
+             '{0}').format(name))
+
+
+def _run_install_script(old_agent, timeout, validate_only=False):
+    # Assuming that if there is no version info in the agent then
+    # this agent was installed by current manager.
+    old_agent = copy.deepcopy(old_agent)
+    old_agent['version'] = old_agent.get('version', _VERSION)
```

Comment:
please use a mechanism similar to 
https://github.com/cloudify-cosmo/cloudify-manager/blob/master/workflows/cloudify_system_workflows/snapshot.py#L166
to extract the version instead of using a hard coded value. This change is unrelated but leaving this code as it is is very problematic